application.tsx

```diff
@@ -343,8 +352,23 @@ export class Application extends ReloadableComponent<{}, ApplicationState> {
     }
   }
 
-  componentDidMount() {
+  async componentDidMount() {
```

Comment:
It's a little weird having this function which is part of React's APIs return a promise. Can you extract the code that does the hash loading into an async method and just invoke it from `componentDidMount`?

Reply:
Fair enough. I've done this in my own code without anything going wrong, but agreed it's a little fishy.